File: backend/src/main/java/br/ufu/facom/petsi/controlaPET/model/Item.java
package br.ufu.facom.petsi.controlaPET.model;

import br.ufu.facom.petsi.controlaPET.model.enums.ItemCondition;
import br.ufu.facom.petsi.controlaPET.model.enums.ItemType;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDate;

@Entity
@Table(name = "items")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Item {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

   @Column(columnDefinition = "TEXT")
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ItemType type;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ItemCondition condition;

    @Column(nullable = false)
    private int totalQuantity;

    @Column(nullable = false)
    private int stockQuantity;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDate createdAt;

    @UpdateTimestamp
    private LocalDate updatedAt;
}

File: frontend/src/interfaces/Item.ts
export type ItemType = 'CONSUMABLE' | 'BORROWABLE' | 'PERMANENT';
export type ItemCondition = 'NEW' | 'GOOD' | 'DAMAGED';

export interface Item {
    id: number
    name: string
    description: string
    type: ItemType
    condition: ItemCondition
    totalQuantity: number
    stockQuantity: number
    createdAt: string
    updatedAt: string
}

export interface CreateItemRequest {
    name: string
    description: string
    type: ItemType
    condition: ItemCondition
    totalQuantity: number
    stockQuantity: number
}

export interface updatedItemRequest {
    name?: string
    description?: string
    type?: ItemType
    condition?: ItemCondition
    totalQuantity?: number
    stockQuantity?: number
}
